diff --git a/core/modules/pgsql/src/Driver/Database/pgsql/Connection.php b/core/modules/pgsql/src/Driver/Database/pgsql/Connection.php
index 041bd43d07c..857be370165 100644
--- a/core/modules/pgsql/src/Driver/Database/pgsql/Connection.php
+++ b/core/modules/pgsql/src/Driver/Database/pgsql/Connection.php
@@ -411,7 +411,7 @@ public function getFullQualifiedTableName($table) {
    *   "mimic_implicit_commit" is used.
    */
   public function addSavepoint($savepoint_name = 'mimic_implicit_commit') {
-    if ($this->inTransaction()) {
+    if ($this->inTransaction() && !$this->transactionManager()->has($savepoint_name)) {
       $this->savepoints[$savepoint_name] = $this->startTransaction($savepoint_name);
     }
   }
