diff --git a/modules/facets_exposed_filters/src/Plugin/views/filter/FacetsFilter.php b/modules/facets_exposed_filters/src/Plugin/views/filter/FacetsFilter.php
index b1fc303..93d0cba 100644
--- a/modules/facets_exposed_filters/src/Plugin/views/filter/FacetsFilter.php
+++ b/modules/facets_exposed_filters/src/Plugin/views/filter/FacetsFilter.php
@@ -138,10 +138,15 @@ class FacetsFilter extends FilterPluginBase {
       );
       $query_type_plugin->build();
 
-      // Skip facet processing and form rendering if there are no results.
-      if (!$facet->getResults()) {
-        return;
-      }
+      // begin islandora-muni
+      // hiding empty facets would change the search form and confuse users
+
+      //// Skip facet processing and form rendering if there are no results.
+      //if (!$facet->getResults()) {
+      //  return;
+      //}
+
+      // end islandora-muni
 
       // Trigger post query stage.
       $processors = $facet->getProcessorsByStage(ProcessorInterface::STAGE_POST_QUERY);
