diff --git a/modules/image/src/Controller/ImageStyleDownloadController.php b/modules/image/src/Controller/ImageStyleDownloadController.php
index 1df6ef1..a7ac6fa 100644
--- a/modules/image/src/Controller/ImageStyleDownloadController.php
+++ b/modules/image/src/Controller/ImageStyleDownloadController.php
@@ -183,6 +183,12 @@ class ImageStyleDownloadController extends FileDownloadController {
       // image exists.
       $path_info = pathinfo(StreamWrapperManager::getTarget($image_uri));
       $converted_image_uri = sprintf('%s://%s%s%s', $this->streamWrapperManager->getScheme($derivative_uri), $path_info['dirname'], DIRECTORY_SEPARATOR, $path_info['filename']);
+
+      // islandora-muni begin
+      // Source file has extension .tiff, not .tiff.png.
+      $converted_image_uri = substr($image_uri, 0, strrpos($image_uri, "."));
+      // islandora-muni end
+
       if (!$this->sourceImageExists($converted_image_uri, $token_is_valid)) {
         $this->logger->notice('Source image at %source_image_path not found while trying to generate derivative image at %derivative_path.', ['%source_image_path' => $image_uri, '%derivative_path' => $derivative_uri]);
         return new Response($this->t('Error generating image, missing source file.'), 404);
