diff --git a/src/Form/OpenIDConnectSettingsForm.php b/src/Form/OpenIDConnectSettingsForm.php
index 2c2ed4c..54569f1 100644
--- a/src/Form/OpenIDConnectSettingsForm.php
+++ b/src/Form/OpenIDConnectSettingsForm.php
@@ -223,7 +223,7 @@ class OpenIDConnectSettingsForm extends ConfigFormBase {
 
       $form['role_mappings'][$role_id] = [
         '#title' => $role->label(),
-        '#type' => 'textfield',
+        '#type' => 'textarea',
         '#default_value' => $default,
       ];
     }
